tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/javainterop/basic/NegativeValidationTest.java

```diff
@@ -118,6 +118,41 @@ public void testMethodNotFound5() {
                 "method_not_found5.bal", 23, 1);
     }
 
+    @Test
+    public void testMethodNotFound6() {
+
+        String path = "test-src/javainterop/negative/method_not_found6.bal";
+
+        CompileResult compileResult = BCompileUtil.compileInProc(path);
+        compileResult.getDiagnostics();
+        Assert.assertEquals(compileResult.getDiagnostics().length, 4);
+        BAssertUtil.validateError(compileResult, 0,
+                "{ballerina/java}METHOD_NOT_FOUND 'No such public method 'getFuture' that matches with parameter " +
+                        "types '(org.ballerinalang.jvm.values.api.BTypedesc)' found in class " +
+                        "'class org.ballerinalang.nativeimpl.jvm.tests.StaticMethods''",
+                "method_not_found6.bal", 3, 1);
+
+        BAssertUtil.validateError(compileResult, 1,
+                "{ballerina/java}METHOD_NOT_FOUND 'No such public method 'getTypeDesc' that matches with parameter " +
+                        "types '(org.ballerinalang.jvm.values.api.BFuture)' found in class " +
+                        "'class org.ballerinalang.nativeimpl.jvm.tests.StaticMethods''",
+                "method_not_found6.bal", 9, 1);
+
+        BAssertUtil.validateError(compileResult, 2,
+                "{ballerina/java}METHOD_NOT_FOUND 'No such public method 'getFutureOnly' that " +
+                        "matches with parameter types " +
+                        "'(org.ballerinalang.jvm.values.api.BFuture,org.ballerinalang.jvm.values.api.BTypedesc)' " +
+                        "found in class 'class org.ballerinalang.nativeimpl.jvm.tests.StaticMethods''",
+                "method_not_found6.bal", 15, 1);
+
+        BAssertUtil.validateError(compileResult, 3,
+                "{ballerina/java}METHOD_NOT_FOUND 'No such public method 'getTypeDescOnly' that " +
+                        "matches with parameter types " +
+                        "'(org.ballerinalang.jvm.values.api.BTypedesc,org.ballerinalang.jvm.values.api.BFuture)' " +
+                        "found in class 'class org.ballerinalang.nativeimpl.jvm.tests.StaticMethods''",
```

Comment:
We can use a String formatter to plug in values for the method name and the  type instead of repeating it multipe time?

```
String s =                 "{ballerina/java}METHOD_NOT_FOUND 'No such public method '%s' that " +
                        "matches with parameter types " +
                        "'(%s)' " +
                        "found in class 'class org.ballerinalang.nativeimpl.jvm.tests.StaticMethods''"
```

Reply:
ack